app/generic/connections/connections.py

```diff
@@ -0,0 +1,249 @@
+import logging
+import os
+from typing import List, Optional
+
+from aries_cloudcontroller import (
+    AcaPyClient,
+    CreateInvitationRequest,
+    InvitationMessage,
+    InvitationRecord,
+    InvitationResult,
+    ReceiveInvitationRequest,
+)
+from aries_cloudcontroller.model.attachment_def import AttachmentDef
+from aries_cloudcontroller.model.invitation_create_request import (
+    InvitationCreateRequest,
+)
+from fastapi import APIRouter, Depends, HTTPException
+from pydantic import BaseModel
+
+from app.dependencies import agent_selector
+from app.generic.connections.models import Connection, conn_record_to_connection
+
+ADMIN_API_KEY = os.getenv("ADMIN_API_KEY", "adminApiKey")
+
+
+logger = logging.getLogger(__name__)
+
+
+router = APIRouter(prefix="/generic/connections", tags=["connections"])
+
+
+class ConnectToPublicDid(BaseModel):
+    public_did: str
+
+
+class CreateInvitation(BaseModel):
+    alias: Optional[str] = None
+    multi_use: Optional[bool] = None
+    use_public_did: Optional[bool] = None
+
+
+class AcceptInvitation(BaseModel):
+    alias: Optional[str] = None
+    use_existing_connection: Optional[bool] = None
+    invitation: ReceiveInvitationRequest
+
+
+class CreateOobInvitation(BaseModel):
+    alias: Optional[str] = None
+    multi_use: Optional[bool] = None
+    use_public_did: Optional[bool] = None
+    attachments: Optional[List[AttachmentDef]] = None
+    create_connection: Optional[bool] = None
+
+
+class AcceptOobInvitation(BaseModel):
+    alias: Optional[str] = None
+    use_existing_connection: Optional[bool] = None
+    invitation: InvitationMessage
+
+
+@router.post("/oob/create-invitation", response_model=InvitationRecord)
+async def create_oob_invitation(
+    body: CreateOobInvitation = CreateOobInvitation(),
+    aries_controller: AcaPyClient = Depends(agent_selector),
+):
+    """
+    Create connection invitation out-of-band.
+    """
+
+    handshake_protocols = [
+        "https://didcomm.org/didexchange/1.0",
+        "https://didcomm.org/connections/1.0",
+    ]
+
+    if not body.create_connection and (
+        not body.attachments or len(body.attachments) == 0
+    ):
+        raise HTTPException(
+            status_code=400,
+            detail="Either or both of 'create_connection' and 'attachments' must be defined / true",
+        )
+
+    oob_body = InvitationCreateRequest(
+        alias=body.alias,
+        attachments=body.attachments,
+        handshake_protocols=handshake_protocols if body.create_connection else None,
+        use_public_did=body.use_public_did,
+    )
+
+    invitation = await aries_controller.out_of_band.create_invitation(
+        multi_use=body.multi_use,
+        body=oob_body,
+        # FIXME: https://github.com/didx-xyz/aries-cloudcontroller-python/issues/67
```

Comment:
ah alright I see - was wondering *weird boolean string*

Reply:
Yep need to fix that. But would like to do a single pass at updating the cloudcontroller and fixing all outstanding bugs